sklearn/cluster/tests/test_bicluster.py

```diff
@@ -231,33 +232,25 @@ def test_perfect_checkerboard():
                            (rows, cols)) == 1
 
 
-def test_errors():
+@pytest.mark.parametrize(
+    "args",
+    [{'n_clusters': (3, 3, 3)}, {'n_clusters': 'abc'},
+     {'n_clusters': (3, 'abc')}, {'method': 'unknown'},
+     {'n_components': 0}, {'n_best': 0},
+     {'svd_method': 'unknown'}]
+)
+def test_errors(args):
     data = np.arange(25).reshape((5, 5))
 
-    model = SpectralBiclustering(n_clusters=(3, 3, 3))
-    assert_raises(ValueError, model.fit, data)
-
-    model = SpectralBiclustering(n_clusters='abc')
-    assert_raises(ValueError, model.fit, data)
-
-    model = SpectralBiclustering(n_clusters=(3, 'abc'))
-    assert_raises(ValueError, model.fit, data)
-
-    model = SpectralBiclustering(method='unknown')
-    assert_raises(ValueError, model.fit, data)
-
-    model = SpectralBiclustering(svd_method='unknown')
-    assert_raises(ValueError, model.fit, data)
-
-    model = SpectralBiclustering(n_components=0)
-    assert_raises(ValueError, model.fit, data)
-
-    model = SpectralBiclustering(n_best=0)
-    assert_raises(ValueError, model.fit, data)
+    model = SpectralBiclustering(**args)
+    with pytest.raises(ValueError):
+        model.fit(data)
 
     model = SpectralBiclustering(n_components=3, n_best=4)
-    assert_raises(ValueError, model.fit, data)
+    with pytest.raises(ValueError):
```

Comment:
Add this one to the parametrization?

Reply:
@NicolasHug Can you explain how I can add this also in parametrization with already exsiting decorator?

Reply:
By adding `{'n_components': 3, 'n_best': 4}` to the list

Reply:
> By adding `{'n_components': 3, 'n_best': 4}` to the list

@NicolasHug I had some trouble making this parameterization. I already have 
```
235 @pytest.mark.parametrize(
236     "args",
237     [{'n_clusters': (3, 3, 3)}, {'n_clusters': 'abc'},
238      {'n_clusters': (3, 'abc')}, {'method': 'unknown'},
239      {'n_components': 0}, {'n_best': 0},
240      {'svd_method': 'unknown'}]
241 )    
242 def test_errors(args):
243     data = np.arange(25).reshape((5, 5))
244 
245     model = SpectralBiclustering(**args)
246     with pytest.raises(ValueError):
247         model.fit(data)
248 
249     model = SpectralBiclustering(n_components=3, n_best=4)
250     with pytest.raises(ValueError):
251         model.fit(data)
252 
253     model = SpectralBiclustering()
254     data = np.arange(27).reshape((3, 3, 3))
255     with pytest.raises(ValueError):
256         model.fit(data)
```
Can you show me how the parameterization can be done?

Reply:
Here is how I would split this test:

```py

@pytest.mark.parametrize(
    "args",
    [{'n_clusters': (3, 3, 3)}, {'n_clusters': 'abc'},
     {'n_clusters': (3, 'abc')}, {'method': 'unknown'},
     {'n_components': 0}, {'n_best': 0},
     {'svd_method': 'unknown'},
     {'n_components': 3, 'n_best': 4}
     ]
)
def test_errors(args):
    data = np.arange(25).reshape((5, 5))

    model = SpectralBiclustering(**args)
    with pytest.raises(ValueError):
        model.fit(data)


def test_wrong_shape():

    model = SpectralBiclustering()
    data = np.arange(27).reshape((3, 3, 3))
    with pytest.raises(ValueError):
        model.fit(data)
```

I added the `     {'n_components': 3, 'n_best': 4}` case as a dict to the parametrization, and created a separate test function for the wrong shape test (it would be repeated for each case if we kept it in the original function)

Reply:
I understand now! Thanks for the nice explanation :smiley: